modules/core/arrow-test/src/main/kotlin/arrow/test/laws/AsyncLaws.kt

```diff
@@ -163,6 +169,40 @@ object AsyncLaws {
         p.equalUnderTheLaw(just(true), EQK)
     }
 
+  fun <F> Async<F>.derivedDefer(GK: GenK<F>, EQK: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.functionToA(GK.genK(Gen.int()))) { f: () -> Kind<F, Int> ->
```

Comment:
I'd use `val f: () -> Kind<F, String> = effect { Thread.currentThread().name }` instead and use `Eq<Kind<F, String>>` this should verify that they behave the same, if the behavior is verified in one of the two.